pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -359,7 +359,7 @@ func (esc *endpointSliceController) InstancesByPort(c *Controller, svc *model.Se
 func (esc *endpointSliceController) newEndpointBuilder(pod *corev1.Pod) *EndpointBuilder {
 	if pod != nil {
 		// Respect pod "istio-locality" label
-		if pod.Labels[model.LocalityLabel] == "" {
+		if pod.Labels != nil && pod.Labels[model.LocalityLabel] == "" {
```

Comment:
we should create the map, not just skip this I think

Reply:
hey @howardjohn, thanks for the feedback. I've updated the PR - let me know what you think.